### PR TITLE
Add fluent API for 2-stream 'mergeLatest'

### PR DESCRIPTION
**Pull request by raboof**
Add fluent API for 2-stream `mergeLatest`

Making it consistent with the other fan-in stages.

Improves https://github.com/akka/akka.github.com/issues/606

Does this make sense? If so I'll add java/scaladocs and do the same for mergePreferred and mergePrioritized

### CODE REVIEW

**Comment by patriknw** [comment]
makes sense, consistency is king

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 1]
```diff
@@ -2594,6 +2594,80 @@ trait FlowOps[+Out, +Mat] {
...
+   * MergeLatest joins elements from N input streams into stream of lists of size N.
+   * i-th element in list is the latest emitted element from i-th input stream.
+   * MergeLatest emits list for each element emitted from some input stream,
+   * but only after each input stream emitted at least one element.
+   *
+   * '''Emits when''' an element is available from some input and each input emits at least one element from stream start
+   *
+   * '''Completes when''' all upstreams complete (eagerClose=false) or one upstream completes (eagerClose=true)
+   */
+  def mergeLatest[U >: Out, M](that: Graph[SourceShape[U], M], eagerComplete: Boolean = false): Repr[List[U]] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala]: `immutable.Seq` instead of `List` in API

**Review by patriknw — COMMENTED** [review]
one thing, but otherwise looking good

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**